### PR TITLE
feat: extend quote fetching logic to include fee [WEB-832]

### PR DESCRIPTION
**Pull request by GabrielBuragev**

### CODE REVIEW

File: packages/shared/src/schemas.ts
[the remark this answers is not in the document]
Comment (GabrielBuragev) [reply packages/shared/src/schemas.ts:1]: Maybe we can postpone its merge for a while?

File: packages/shared/src/functions.ts [hunk 1]
```diff
@@ -13,3 +13,8 @@ export const bigintMin = (...args: bigint[]): bigint =>
 
 export const bigintMax = (...args: bigint[]): bigint =>
   args.reduce((max, current) => (current > max ? current : max));
+
+export const calculateBpsAmountFromTotalAmount = (
```
Comment (zoheb391) [inline packages/shared/src/functions.ts:17]:
```suggestion
export const calculateBpsAmountFromAmount = (
```

File: packages/sdk/src/swap/types.ts [hunk 2]
```diff
@@ -31,16 +31,16 @@ interface ChainsAndAssets {
 
 export interface QuoteRequest extends ChainsAndAssets {
   amount: string;
+  boostFeeBps?: number;
```
Comment (niklasnatter) [inline packages/sdk/src/swap/types.ts:34]: did we forget to add `brokerCommissionBps` here?

Comment (niklasnatter) [reply packages/sdk/src/swap/types.ts:34]: okay the broker fee is read from the config, so i guess it makes sense 🤔

File: packages/shared/src/functions.ts [hunk 3]
```diff
@@ -13,3 +13,6 @@ export const bigintMin = (...args: bigint[]): bigint =>
 
 export const bigintMax = (...args: bigint[]): bigint =>
   args.reduce((max, current) => (current > max ? current : max));
+
+export const calculateBpsAmountFromAmount = (bps: bigint, amount: bigint) =>
```
Comment (niklasnatter) [inline packages/shared/src/functions.ts:17]: we have something like this in `packages/swap/src/utils/fees.ts`, maybe we can merge them

**Review by niklasnatter — APPROVED** [review]

File: packages/shared/src/functions.ts (hunk 3 above)
Comment (GabrielBuragev) [reply packages/shared/src/functions.ts]: Good catch. I didn't see the existence of the other function.
Moved it to shared as well as the test for it.

Comment (GabrielBuragev) [reply packages/shared/src/functions.ts]: just realized that they are different functions. Looking into how i can merge them

**Review by niklasnatter — APPROVED** [review]

**Pull request merged**